hls4ml/templates/vivado_template.py

```diff
@@ -399,9 +479,9 @@ def __init__(self):
         self.register_templates('Resize'                 , resize_function_template,      resize_config_template, resize_include_list)
         self.register_templates('Transpose'              , transpose_function_template,   transpose_config_template, transpose_include_list)
         self.register_templates('GarNet'                 , garnet_function_template,      garnet_config_template, garnet_include_list)
-        self.register_templates('GarNetStack'            , garnet_stack_function_template,garnet_stack_config_template, garnet_include_list)        
-    
-    def get_valid_reuse_factors(self, layer):
+        self.register_templates('GarNetStack'            , garnet_stack_function_template,garnet_stack_config_template, garnet_include_list)    
+
+    def get_valid_reuse_factors(self, layer, recr = False):
```

Comment:
I mentioned this before, I think you can get away without `recr` parameter. Or refactor this function so that it accepts `n_in` and `n_out` parameters and move their computation to the caller